bfd/src/main/java/gov/cms/ab2d/bfd/client/BFDSearchImpl.java

```diff
@@ -0,0 +1,75 @@
+package gov.cms.ab2d.bfd.client;
+
+import ca.uhn.fhir.parser.IParser;
+import ca.uhn.fhir.rest.server.exceptions.ResourceNotFoundException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+
+import java.io.InputStream;
+import java.time.OffsetDateTime;
+
+@Component
+@Slf4j
+public class BFDSearchImpl implements BFDSearch {
+
+    private final HttpClient httpClient;
+
+    private final IParser parser;
+
+    private final Environment environment;
+
+    @Value("${bfd.serverBaseUrl}")
+    private String serverBaseUrl;
+
+    public BFDSearchImpl(HttpClient httpClient, IParser iParser, Environment environment) {
+        this.httpClient = httpClient;
+        this.parser = iParser;
+        this.environment = environment;
+    }
+
+    @Override
+    public Bundle searchEOB(String patientId, OffsetDateTime since, int pageSize) throws IOException {
+        StringBuilder url = new StringBuilder(serverBaseUrl + "ExplanationOfBenefit?patient=" + patientId + "&excludeSAMHSA=true");
+
+        if (since != null) {
+            url.append("&_lastUpdated=ge").append(since);
+        }
+
+        if (pageSize > 0) {
+            url.append("&_count=").append(pageSize);
+        }
+
+        HttpGet request = new HttpGet(url.toString());
+        // No active profiles means use JSON
+        if (environment.getActiveProfiles().length == 0) {
+            request.addHeader("Accept", "application/fhir+json;q=1.0, application/json+fhir;q=0.9");
+        }
+
+        request.addHeader("Accept-Encoding", "gzip");
+        request.addHeader("Accept-Charset", "utf-8");
+
+        try (CloseableHttpResponse response = (CloseableHttpResponse) httpClient.execute(request)) {
+            int status = response.getStatusLine().getStatusCode();
+            if (status >= 200 && status < 300) {
```

Comment:
What other 2xx codes are expected?  Will this code properly handle those situations?

Use already defined constants instead of literals.

A switch statement would be more direct instead of a series of if/elseif/elseif/else.

Reply:
No need to worry about other 200 codes, we just care about a success

Reply:
Switch makes more sense in an enum for type safety to exhaust all options, it's no different in this case.